packages/flutter_driver/lib/src/common/wait.dart

```diff
@@ -0,0 +1,267 @@
+  const WaitUntilNoTransientCallbacks({ Duration timeout }) : super(timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  WaitUntilNoTransientCallbacks.deserialize(Map<String, String> json)
+      : super.deserialize(json);
+
+  @override
+  String get kind => 'waitUntilNoTransientCallbacks';
+}
+
+/// A Flutter Driver command that waits until the frame is synced.
+///
+/// This command has been deprecated in favor of [WaitForCondition]. Construct
+/// a command that waits until no pending frame as follows:
+///
+/// ```dart
+/// WaitForCondition noPendingFrame = WaitForCondition(NoPendingFrame());
+/// ```
+@Deprecated('This command has been deprecated in favor of WaitForCondition. '
+            'Use WaitForCondition command with NoPendingFrame.')
+class WaitUntilNoPendingFrame extends Command {
+  /// Creates a command that waits until there's no pending frame scheduled.
+  const WaitUntilNoPendingFrame({ Duration timeout }) : super(timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  WaitUntilNoPendingFrame.deserialize(Map<String, String> json)
+      : super.deserialize(json);
+
+  @override
+  String get kind => 'waitUntilNoPendingFrame';
+}
+
+/// A Flutter Driver command that waits until the Flutter engine rasterizes the
+/// first frame.
+///
+/// {@template flutter.frame_rasterized_vs_presented}
+/// Usually, the time that a frame is rasterized is very close to the time that
+/// it gets presented on the display. Specifically, rasterization is the last
+/// expensive phase of a frame that's still in Flutter's control.
+/// {@endtemplate}
+///
+/// This command has been deprecated in favor of [WaitForCondition]. Construct
+/// a command that waits until no pending frame as follows:
+///
+/// ```dart
+/// WaitForCondition firstFrameRasterized = WaitForCondition(FirstFrameRasterized());
+/// ```
+@Deprecated('This command has been deprecated in favor of WaitForCondition. '
+            'Use WaitForCondition command with FirstFrameRasterized.')
+class WaitUntilFirstFrameRasterized extends Command {
+  /// Creates this command.
+  const WaitUntilFirstFrameRasterized({ Duration timeout }) : super(timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  WaitUntilFirstFrameRasterized.deserialize(Map<String, String> json)
+      : super.deserialize(json);
+
+  @override
+  String get kind => 'waitUntilFirstFrameRasterized';
+}
+
+/// Thrown to indicate a serialization error.
+class SerializationException implements Exception {
+  /// Creates a [SerializationException] with an optional error message.
+  const SerializationException([this.message]);
+
+  /// The error message, possibly null.
+  final String message;
+
+  @override
+  String toString() => 'SerializationException($message)';
+}
+
+/// Base class for Flutter Driver wait conditions, objects that describe conditions
+/// the driver can wait for.
```

Comment:
This should document that if you subclass this, you also need to implement a `WaitCondition` in the extension. Also document the relationship between the two.

Reply:
Done. PTAL.